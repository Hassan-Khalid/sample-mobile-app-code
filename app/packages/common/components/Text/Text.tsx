import {Text as NText,TextStyle, TextProps} from "react-native"
import React from "react";
import {useTranslation} from 'react-i18next';
import { AnyType } from 'common/declarations/types';
import AppColors from 'common/assets/colors/AppColors';

interface _TextProps extends TextProps{
  style?:TextStyle,
  children:React.ReactNode | AnyType,
}

const Text = (props:_TextProps)=>{
  const {t} = useTranslation();
  return(
    <NText style={{...props.style,...{color:AppColors.BACKGROUND}}} {...props} >
      {t(props.children)}
    </NText>
  )
}

export default Text;