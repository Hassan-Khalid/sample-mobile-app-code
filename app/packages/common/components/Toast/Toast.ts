import React from 'react'

import { useToast } from "react-native-toast-notifications";
import { ToastProps } from '../../declarations/types';
import { t } from 'i18next';
const Toast = ({message,type,
                 placement='top',
                 duration =2000,
                 animationType ='slide-in',
               }:ToastProps) => {
  const toast = useToast()
  return toast.show(t(message), {
    type: type,placement: placement,duration: duration, animationType: animationType
  });
  
}

export default Toast