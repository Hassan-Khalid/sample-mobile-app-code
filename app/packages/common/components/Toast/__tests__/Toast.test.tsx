import React from 'react';
import { render } from '@testing-library/react-native';
import Toast from '../Toast';
import { useToast } from 'react-native-toast-notifications';
import Text from '../../Text/Text';

// Mock the useToast hook
jest.mock('react-native-toast-notifications', () => ({
  useToast: () => ({
    show: jest.fn(),
  }),
}));

describe('Toast component', () => {
  it('displays the toast notification with the correct message and options', () => {
    // Arrange
    const message = 'Test message';
    const type = 'success';
    const placement = 'top';
    const duration = 2000;
    const animationType = 'slide-in';
    
    // Act
    const wrapper = render(
      <Toast
        message={message}
        type={type}
        placement={placement}
        duration={duration}
        animationType={animationType}
      />);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
    
  });
});
