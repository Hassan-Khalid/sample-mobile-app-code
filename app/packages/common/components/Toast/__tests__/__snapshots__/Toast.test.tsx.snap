// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component displays the toast notification with the correct message and options 1`] = `null`;
